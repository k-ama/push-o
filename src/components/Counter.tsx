import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Actions } from '../store/actions'
import './Counter.css';
import Color from '../types/Color'
import { computeRBGTarget } from '../utils/utils'


export const Counter = (props: {minColor: Color, maxColor: Color}) => {
  const dispatch = useDispatch()
  const selector = (state: any) => {
    return state.count
  }
  const count = useSelector(selector)


  const countTimeout = setTimeout(() => {
    if (count > 0) {
      dispatch(Actions.timeout())
    }
  }, 1000)
  const onCountChange = () => {
    if (count < 100) {
      clearTimeout(countTimeout)
      dispatch(Actions.userPush())
    }

 }

  const {
    r,
    g,
    b
  } = computeRBGTarget(props.minColor, props.maxColor, count)

  return (
    <div className="Counter">
      <div
        className="Counter-button"
        style={{backgroundColor:`rgb(${r},${g},${b})`}} // eslint-disable-line
        onClick={onCountChange}>
      </div>
    </div>
  );
}

export default Counter;
