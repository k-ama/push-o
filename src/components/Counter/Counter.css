.Counter{
  flex-grow: 1;
}
.Counter .Counter-button{
  width: 100%;
  height: 100%;
  border: 0;
  transition: background-color 2s;
}
