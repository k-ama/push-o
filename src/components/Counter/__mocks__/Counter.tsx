import React from "react"
import { ReactElement } from "react"

const CounterMock = (): ReactElement => {
  return <div className="Counter"/>
}

export default CounterMock