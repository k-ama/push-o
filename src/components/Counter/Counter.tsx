import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Actions } from '../../store/actions'
import './Counter.css';
import Color from '../../types/Color'
import { computeRBGTarget } from '../../utils/utils'
import { toggleStateSelector, countSelector } from '../../store/selector'


export const Counter = (props: { minColor: Color, maxColor: Color}) => {
  const dispatch = useDispatch()
  const count = useSelector(countSelector)
  const toggleState = useSelector(toggleStateSelector)

  const countTimeout = setTimeout(() => {
    if (!toggleState) {
      return
    }
    if (count > 0) {
      dispatch(Actions.timeout())
    }
  }, 1000)
  const onCountChange = () => {
    if (!toggleState) {
      return
    }
    if (count < 100) {
      clearTimeout(countTimeout)
      dispatch(Actions.userPush())
    }
 }

  const {
    r,
    g,
    b
  } = computeRBGTarget(props.minColor, props.maxColor, count)

  return (
    <div className="Counter">
      <div
        className="Counter-button"
        style={{backgroundColor:`rgb(${r},${g},${b})`}} // eslint-disable-line
        onClick={onCountChange}>
      </div>
    </div>
  );
}

export default Counter;
