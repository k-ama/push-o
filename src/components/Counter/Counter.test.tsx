import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Counter from './Counter'

describe('Counter', () => {

  describe('Snapshot', () => {
    test('renders Counter', () => {
      const mockStore = configureStore()
      const minColor = {
        r:0,
        g:0,
        b:0
      }
      const maxColor = {
        r:100,
        g:100,
        b:100
      }
      const initialState = {
        count:0
      }
      let store = mockStore(initialState)
      const component = renderer.create(<Provider store={store}><Counter minColor={minColor} maxColor={maxColor}/></Provider>)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })


})
