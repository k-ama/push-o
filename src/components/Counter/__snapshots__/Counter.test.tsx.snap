// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot renders Counter 1`] = `
<div
  className="Counter"
>
  <div
    className="Counter-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(0,0,0)",
      }
    }
  />
</div>
`;
